access_helpers.py

```diff
@@ -13,34 +13,50 @@
 import base64
 import json
 import boto3
-import cachetools.func
 import os
+from cachetools import cached, TTLCache
+from threading import Lock
+
 
 SSM_PARAMETER_NAME = os.environ.get('SYNAPSE_TOKEN_AWS_SSM_PARAMETER_NAME')
 SSM_KMS_KEY_ALIAS = os.environ.get('KMS_KEY_ALIAS')
 AWS_REGION = os.environ.get('AWS_REGION')
 EC2_INSTANCE_ID = os.environ.get('EC2_INSTANCE_ID')
 CACHE_TTL_SECONDS = 3600 # one hour
 
+APPROVED_USER_LOCK=Lock()
+STORE_TO_SSM_LOCK=Lock()
+GET_PUBLIC_KEY_LOCK=Lock()
+
 def get_approved_user_from_ec2_instance_tags(tags):
   for tag in tags:
     if tag["Key"] == 'Protected/AccessApprovedCaller':
       approved_caller = tag["Value"]
 
   return approved_caller.split(':')[1] #return userid portion of tag
 
-
-
-@cachetools.func.ttl_cache(maxsize=1, ttl=CACHE_TTL_SECONDS)
+# cachetools is not thread safe so we implement thread safety ourselves
 def approved_user():
```

Comment:
Is this wrapper equivalent to using the lock parameter in @cached?

Reply:
@xschildw According to the doc's, the lock parameter should do the same thing however it can be demonstrated that it does not work.